tests/CSSList/DocumentTest.php

```diff
@@ -85,4 +85,62 @@ public function setContentsReplacesContentsSetInPreviousCall(): void
 
         self::assertSame($contents2, $this->subject->getContents());
     }
+
+    /**
+     * @return array<string, array<string, DeclarationBlock|array<int, DeclarationBlock>>>
+     */
+    public static function insertContentBeforeInsertsContentBeforeSibblingOrAppendsIfSibblingNotFoundDataProvider(): array
+    {
+
+        $bogusOne = new DeclarationBlock();
+        $bogusOne->setSelectors('.bogus-one');
+        $bogusTwo = new DeclarationBlock();
+        $bogusTwo->setSelectors('.bogus-two');
+
+        $oItem = new DeclarationBlock();
+        $oItem->setSelectors('.item');
+
+        $oSibling = new DeclarationBlock();
+        $oSibling->setSelectors('.sibling');
+
+        $oOrphan = new DeclarationBlock();
+        $oOrphan->setSelectors('.forever-alone');
+
+        return [
+            'insert before' => [
+                'initialContent' => [$bogusOne, $oSibling, $bogusTwo],
+                'oItem' => $oItem,
+                'oSibling' => $oSibling,
+                'expectedContent' => [$bogusOne, $oItem, $oSibling, $bogusTwo],
+            ],
+            'append if not found' => [
+                'initialContent' => [$bogusOne, $oSibling, $bogusTwo],
+                'oItem' => $oItem,
+                'oSibling' => $oOrphan,
+                'expectedContent' => [$bogusOne, $oSibling, $bogusTwo, $oItem],
+            ]
+        ];
+    }
+
+    /**
+     * @test
+     *
+     * @dataProvider insertBeforeDataProvider
+     */
+    public function insertContentBeforeInsertsContentBeforeSibblingOrAppendsIfSibblingNotFound(
```

Comment:
Let's split this into two tests. (The "or" in the test name is a sign that this test wants to be two tests as it tests two different pieces of behavior.)